src/ios/getHeaderSearchPath.js

```diff
@@ -0,0 +1,38 @@
+const path = require('path');
+const union = require('lodash.union');
+
+/**
+ * Returns last item from an array
+ */
+const lastItem = (array) => array[array.length - 1];
+
+/**
+ * Given an array of headers it returns search path so Xcode can find them
+ * If there are multiple header files located accross different folders,
+ * it finds the top-most one and returns recursive version of it
+ */
+module.exports = function getHeaderSearchPath(sourceDir, headers) {
+  const directories = union(
+    headers.map(path.dirname)
+  );
+
+  if (directories.length === 1) {
+    return `"$(SRCROOT)/${path.relative(sourceDir, directories[0])}"`;
```

Comment:
if `directories[0]` is a `targetDir`, I'd like to add a line above declaring it. Assumptions in the code is always bad.